src/lib.rs

```diff
@@ -69,3 +91,43 @@ pub fn include_glsl(tokens: TokenStream) -> TokenStream {
     let IncludeGlsl(output) = parse_macro_input!(tokens as IncludeGlsl);
     output.expand()
 }
+
+/// Compile inline GLSL source
+///
+/// ```
+/// use vk_shader_macros::glsl;
+/// const VERT: &[u32] = glsl! {
+///     version: 450, kind: vert, optimize: size, target: vulkan1_1;
+///     r#"
+/// #version 450
+///
+/// void main() {
+///     gl_Position = vec4(0);
+/// }
+/// "#
+/// };
+/// ```
+///
+/// # Limitations
+///
+/// - Because the shader kind cannot be inferred from a file extension,
+///   you may need to specify it manually as the above example does.
+/// - Due to limitations of proc macros, relative includes cannot be resolved
+///   relative to the macro invocation. Standard includes must be used instead:
+///
+///   ```glsl
+///   // standard
+///   #include <crate/to/include.glsl>
```

Comment:
This example is confusing, since it suggests that the crate itself must be part of the path. Maybe drop the code example and just specify that includes will always be resolved relative to the crate root?